client/domain/FeedbackObjectFactorySpec.js

```
@@ -0,0 +1,55 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for FeedbackObject domain objects.
+ */
+
+describe('FeedbackObjectFactory', function() {
+  var FeedbackObjectFactory;
+
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    FeedbackObjectFactory = $injector.get(
+      'FeedbackObjectFactory');
+  }));
+
+  describe('isAnswerCorrect', function() {
+    it('should correctly get whether answer is correct', function() {
+      var Feedback = FeedbackObjectFactory.create(true);
+      expect(Feedback.isAnswerCorrect()).toBe(true);
+    });
+  });
+
+  describe('firstParagraphText', function() {
+    it('should error when adding first paragraph as code', function() {
```

should throw an error when ...